DiscordAlertsTrader/brokerages/ibkr_api.py

```diff
@@ -0,0 +1,525 @@
+            kwargs['action'] = "SELL"
+        elif action == "BTC":
+            kwargs['action'] = "BUY"
+
+        Symbol = self.fix_symbol(Symbol, "in")
+        if "_" in Symbol:
+            kwargs['asset'] = 'OPT'
+        else:
+            kwargs['asset'] ='STK'
+            kwargs['outsideRegularTradingHour'] = False
+            kwargs['stock'] = Symbol
+
+        kwargs['enforce'] ='GTC'
+        kwargs['quant'] = Qty 
+        kwargs['orderType'] = 'STP LMT' 
+        kwargs['stpPrice'] = SL
+        kwargs['lmtPrice'] = SL
+        kwargs['conId'] = self.get_con_id(Symbol)
+
+        return kwargs if kwargs['conId'] is not None else None
+
+    def make_STC_SL_trailstop(self, Symbol:str, Qty:int,  trail_stop_const:float, action="STC", **kwarg):
+        "trail_stop_const"
+        kwargs = {}
+        if action == "STC":
+            kwargs['action'] = "SELL"
+        elif action == "BTC":
+            kwargs['action'] = "BUY"
+
+        Symbol = self.fix_symbol(Symbol, "in")
+        if "_" in Symbol:
+            kwargs['asset'] = 'OPT'
+        else:
+            kwargs['asset'] ='STK'
+            kwargs['outsideRegularTradingHour'] = True
+            kwargs['stock'] = Symbol    
+
+        kwargs['enforce'] ='GTC'
+        kwargs['quant'] = Qty 
+        kwargs['orderType'] = 'TRAIL'
+        kwargs['trial_value'] = trail_stop_const
+        kwargs['trial_type'] = 'DOLLAR'
+        kwargs['outsideRegularTradingHour'] = True
+        kwargs['conId'] = self.get_con_id(Symbol)
+
+        return kwargs if kwargs['conId'] is not None else None
+    
+    def get_quotes(self, symbol:list):
+        
+        quotes = {}
+        for symbol in symbols:
+            con_id = self.get_con_id(symbol)
+            print(con_id)
+            contract = Contract(conId=con_id)
+            contract = self.ib.qualifyContracts(contract)[0]
+            self.ib.sleep(0.1)
+            quote = self.ib.reqTickers(contract)
+            self.ib.sleep(0.1)
+            quotes[symbol] = {
+                'symbol': symbol,
+                'mid': ((quote[-1].ask + quote[-1].bid) / 2) if quote[-1].ask and quote[-1].bid else float('nan'),
+                'bid': quote[-1].bid,
+                'ask': quote[-1].ask,
+                'quoteTimeInLong': int(round(quote[-1].time.timestamp())) if quote[-1].time else None,
+            }
+        
+        return quotes
+    
+    def _convert_option_fromts(self, ticker):
+        """
+        Convert ticker from 'NFLX 051724C450' to 
+
+        {
+            'symbol':'NFLX',
+            'lastTradeDateOrContractMonth':'20240517',
+            'strike': 450,
+            'right':'C',
+        }
+        
+        Parameters:
+        ticker (str): Ticker in the original format.
+        
+        Returns:
+        dict: kwargs in desired format.
+        """
+        if "_" not in ticker:
+            return ticker
+        symb, option_part = ticker.split(" ")  # Split the ticker by spaces
```

Comment:
IBKR has spaces in the ticker??? are you sure you changed that example from tradestation?

Reply:
should  be _convert_option_fromIBKR

Reply:
Actually IBKR has the option as a Contract Object https://ib-insync.readthedocs.io/api.html#module-ib_insync.contract

I didn't realise it was meant to do that